### PR TITLE
(PC-16053) fix(Search): SearchBox query and NoResults handled with URL

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-16053

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/search/pages/__tests__/Search.native.test.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ import { SuggestedVenue } from 'libs/venue'
 import { mockedSuggestedVenues } from 'libs/venue/fixtures/mockedSuggestedVenues'
 import { render, fireEvent } from 'tests/utils'
 
+import { useRoute } from '../../../../../__mocks__/@react-navigation/native'
```
Comment (aliraiki) [inline src/features/search/pages/__tests__/Search.native.test.tsx:14]:
```suggestion
import { useRoute } from '__mocks__/@react-navigation/native'
```

File: src/features/search/atoms/__tests__/NoSearchResult.web.test.tsx [hunk 2]
```diff
@@ -1,17 +1,20 @@
 import React from 'react'
 
+import { LocationType } from 'features/search/enums'
 import { initialSearchState } from 'features/search/pages/reducer'
+import { MAX_RADIUS } from 'features/search/pages/reducer.helpers'
 import { analytics } from 'libs/firebase/analytics'
 import { GeoCoordinates } from 'libs/geolocation'
 import { fireEvent, render } from 'tests/utils/web'
 
+import { push, useRoute } from '../../../../../__mocks__/@react-navigation/native'
```
Comment (aliraiki) [inline src/features/search/atoms/__tests__/NoSearchResult.web.test.tsx:10]:
```suggestion
import { push, useRoute } from '__mocks__/@react-navigation/native'
```

**Review by aliraiki — APPROVED** [review]

File: src/features/search/pages/useNavigateWithStagedSearch.ts [hunk 3]
```diff
@@ -11,10 +12,11 @@ export const useNavigateWithStagedSearch = () => {
   const { push } = useNavigation<UseNavigationType>()
 
   return useCallback(
-    (partialSearchState?: Partial<SearchState>) => {
+    (partialSearchState?: Partial<SearchState>, reset?: boolean) => {
```
Comment (voisinhugo) [inline src/features/search/pages/useNavigateWithStagedSearch.ts:15]: Ça vaut peut-être le coup de demander des arguments nommés : c'est pas hyper clair d'avoir un `true` qui traine, alors que `{partialSearchState: blablabla, reset: true}` c'est déjà plus limpide à la lecture

Comment (kopax-polyconseil) [reply src/features/search/pages/useNavigateWithStagedSearch.ts:15]: Pas sur de comprendre ton commentaire, comme le 2nd arguments est nommé `reset.

**Pull request merged**